Documentation/network/kubernetes/configuration.rst

```diff
@@ -150,65 +148,60 @@ The CNI auto installation is performed as follows:
 1. The ``/etc/cni/net.d`` and ``/opt/cni/bin`` directories are mounted from the
    host filesystem into the pod where Cilium is running.
 
-2. The file ``/etc/cni/net.d/05-cilium.conf`` is written in case it does not
+2. The file ``/etc/cni/net.d/05-cilium.conflist`` is written in case it does not
    exist yet.
 
 3. The binary ``cilium-cni`` is installed to ``/opt/cni/bin``. Any existing
    binary with the name ``cilium-cni`` is overwritten.
 
-.. _install_cni:
-
-Manually installing CNI
------------------------
-
-This step is typically already included in all Kubernetes distributions or
-Kubernetes installers but can be performed manually:
-
-.. code-block:: shell-session
-
-    sudo mkdir -p /opt/cni
-    wget https://storage.googleapis.com/kubernetes-release/network-plugins/cni-0799f5732f2a11b329d9e3d51b9c8f2e3759f2ff.tar.gz
-    sudo tar -xvf cni-0799f5732f2a11b329d9e3d51b9c8f2e3759f2ff.tar.gz -C /opt/cni
-    rm cni-0799f5732f2a11b329d9e3d51b9c8f2e3759f2ff.tar.gz
-
-
 Adjusting CNI configuration
 ---------------------------
 
 The CNI configuration file is automatically written and maintained by the
-scripts ``cni-install.sh`` and ``cni-uninstall.sh`` which are running as
-``postStart`` and ``preStop`` hooks of the Cilium pod.
-
-If you want to provide your own custom CNI configuration file, set the
-``CILIUM_CUSTOM_CNI_CONF`` environment variable to avoid overwriting the
-configuration file by adding the following to the ``env:`` section of the
-``cilium`` DaemonSet:
-
-.. code-block:: yaml
-
-        - name: CILIUM_CUSTOM_CNI_CONF
-          value: "true"
-
-The CNI installation can be configured with environment variables. These
-environment variables can be specified in the :term:`DaemonSet` file like this:
-
-.. code-block:: yaml
+cilium pod. It is written after the agent has finished initialization and
+is ready to handle pod sandbox creation. In addition, the agent will remove
+any other CNI configuration files by default.
+
+There are a number of Helm variables that adjust CNI configuration management.
+For a full description, see the helm documentation. A brief summary:
+
++----------------+----------------------------------------+---------+
+| Helm variable  | Description                            | Default |
++================+========================================+=========+
+| cni.customConf | Disable CNI configuration management   | false   |
```

Comment:
nit: this feels like it should contain some custom conf object, not a boolean. Maybe something like `cni.useCustomConf`?

Reply:
I didn't change the semantics of any helm values, I just documented them.

Yes, the API could have been a bit better thought out, but what we nave now works fine enough.